### PR TITLE
Make BreadCrumbControl to UIScrollView

### PR DESCRIPTION
**Pull request by rikusouda**
If BreadCrumbControl's items are too long, it needs scrolling. 
No longer use `self.sendActions( for: UIControlEvents.touchUpInside)`, so I provide `BreadCrumbControlDelegate` to notify button touch up.
And I added `autoScrollEnabled` option. If it is true, scroll automatically if last item is over frame.

### CODE REVIEW

**Comment by apparition47** [comment]
I used this project in a personal app and I embeded it in a `ScrollView` for when the BreadCrumbControl's items are too long--never thought to implement it directly in this project. So thanks again for your contribution. 

But might I suggest a few things? 

1. Maybe have the scroll indicator an optional property?
screenshot_2017_08_24_22_51


2. Make `autoScrollEnable` a `@IBInspectable ` property?
main_storyboard_ _edited

**Comment by rikusouda** [comment]
I'm glad to get your suggestions.

> Maybe have the scroll indicator an optional property?

I think, you can make disable it with `UIScrollView.showsHorizontalScrollIndicator`. Do you need another option?
Do you want that `UIScrollView.showsHorizontalScrollIndicator` default value is false.

> Make autoScrollEnable a @IBInspectable property?

Sure. I'll do it.

**Comment by rikusouda** [comment]
Made autoScrollEnable a @IBInspectable property. And changed sample.

File: BreadCrumb Control/BreadCrumb Control/BreadCrumb.swift [hunk 1]
```diff
@@ -62,14 +70,14 @@ public class CBreadcrumbControl: UIControl{
         NotificationCenter.default.addObserver(self, selector: #selector(self.receivedUINotificationNewItems), name:NSNotification.Name(rawValue: "NotificationNewItems"), object: nil)
     }
     
+    @IBInspectable public var autoScrollEnabled = false
```
Comment (apparition47) [inline BreadCrumb Control/BreadCrumb Control/BreadCrumb.swift]: For whatever reason in XCode (using 8.3.3), `@IBInspectable` properties don't seem to show up in the Attributes Inspector unless you explicitly define its type (i.e. `Bool`).

`@IBInspectable public var autoScrollEnabled: Bool = false`

**Review by apparition47 — CHANGES_REQUESTED** [review]

**Review by apparition47 — APPROVED** [review]

**Comment by apparition47** [comment]
Thanks!

File: BreadCrumb Control/BreadCrumb Control/BreadCrumb.swift (hunk 1 above)
Comment (rikusouda) [reply BreadCrumb Control/BreadCrumb Control/BreadCrumb.swift]: I defined `autoScrollEnabled` type explicitly for Interface Builder. And confirmed it on XCode 8.3.3.
autoscrollenabled

**Pull request merged**